pallene/function_inline.lua

```diff
@@ -0,0 +1,179 @@
+-- Copyright (c) 2020, The Pallene Developers
+-- Pallene is licensed under the MIT license.
+-- Please refer to the LICENSE and AUTHORS files for details
+-- SPDX-License-Identifier: MIT
+
+local ir = require "pallene.ir"
+local util = require "pallene.util"
+
+local function_inline = {}
+
+
+-- return moves corresponding to each argument as they where at the beginning of the inner scope
+local function args_to_moves(args,cmd,top)
+    local arguments = {}
+    for k,_ in ipairs(args) do
+        -- in PIR, the variable name convention is arg1 -> x1, arg2 -> x2 ... argN -> xN
+        -- and then local and temps use values starting from N+1
+        -- so we can just use the srcs from the call (the arguments) on the rhs of the 'move' commands
+        -- at the same time, it is also convenient to adjust the dst (lhs) of the 'move'
+        -- to values that will not collide with the outer scope (adding top)
```

Comment:
This explanation went right over my head. Maybe it would be easier to explain with a concrete example instead?
Perhaps by pointing to one of the test cases?